src/SymfonyCache/KernelDispatcher.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\SymfonyCache;
+
+use FOS\HttpCache\Exception\ExceptionCollection;
+use FOS\HttpCache\ProxyClient\Dispatcher;
+use Psr\Http\Message\RequestInterface;
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+/**
+ * An implementation of Dispatcher that allows direct calling of the
+ * Symfony HttpCache kernel without executing a real HTTP request.
+ * It can only be used if you have a single node setup of Symfony and serves
+ * as kind of a shortcut for easier configuration.
+ * If you use Varnish or have a multiple node Symfony setup, this client is entirely
+ * useless and cannot be used.
+ *
+ * @author Yanick Witschi <yanick.witschi@terminal42.ch>
+ */
+class KernelDispatcher implements Dispatcher
+{
+    /**
+     * @var HttpCacheProvider
+     */
+    private $kernel;
+
+    /**
+     * @var array
+     */
+    private $queue = [];
+
+    /**
+     * KernelClient constructor.
+     *
+     * @param HttpCacheProvider $kernel
+     */
```

Comment:
please remove this autogenerated phpdoc, it does not add anyhting thats not in the code already.